lib/elixir/src/elixir_tokenizer.erl

```diff
@@ -129,10 +129,16 @@ tokenize(String, Line, Column, Opts) ->
     false -> true
   end,
 
+  Preserve = case lists:keyfind(preserve_comments, 1, Opts) of
+    {preserve_comments, true} -> true;
+    false -> false
+  end,
```

Comment:
This will fail for the option `preserve_comments: false` but I guess we have this issue in the code above also.